### PR TITLE
Stop cleanup_jobs crashing on a batch window that holds no jobs

### PR DESCRIPTION
**Pull request by blaipr**
## Problem

`cleanup_jobs` walks the id range in batches and adds up what each batch removed:

```python
_, results = qs_batch.delete()
deleted += results['main.Job']
```

Django returns `(0, {})` when a delete matches nothing, so an empty window raises `KeyError` and the command dies:

```
empty Job delete -> (0, {})
results['main.Job'] -> KeyError 'main.Job'
```

Driving the real command with a gap in the id range:

```
job ids [929, 930, 931, 932, 933, 934], deleted [931, 932] to leave a gap
cleanup_jobs -> CRASHED with KeyError 'main.Job'
```

Two ways to land on an empty window, both from the loop at `cleanup_jobs.py:304`:

- **Boundary overlap.** A window is `[start, start + batch_size]` and the next one begins at `start + batch_size`, so they share an id. The last window starts at `min + k*batch_size`; when the id span is an exact multiple of the batch size, the highest job is already removed by the previous window and the final window is empty.
- **Sparse id ranges.** `Job` shares the `UnifiedJob` id sequence with project updates, inventory updates and workflow jobs, so a window covering only ids belonging to other job types holds no deletable `Job`. That gets likelier the smaller the batch, and `--batch-size` is a documented option (`cleanup_jobs.py:191`).

This matters more than a management-command traceback suggests. `cleanup_jobs` is the built-in "Remove jobs older than a certain number of days" system job (`jobs.py:1253`), normally on a schedule, so a failure stops job cleanup and lets the database grow.

## Fix

Use `results.get('main.Job', 0)`. The sibling helper `_delete_unpartitioned_events` already guards its own empty list the same way at line 283, which is what makes this look like an oversight rather than an assumption.

## Tests

Three tests in `awx/main/tests/functional/commands/test_cleanup_jobs.py`: the gap case, the exact-multiple boundary case, and a control that deletes everything in one window.

Before the fix, the first two fail with the real `KeyError: 'main.Job'` and the control passes. After, all pass along with the existing unit tests:

```
py.test -q awx/main/tests/functional/commands/test_cleanup_jobs.py awx/main/tests/unit/commands/test_cleanup_jobs.py
............                                         [100%]
12 passed in 1.15s
```

One thing worth flagging separately: the tests have to stub `_pre_delete_job_host_summaries` and `has_unpartitioned_table`. Both issue PostgreSQL-only SQL (`ANY(%s)` and a read of `pg_tables`) while the suite runs against SQLite (`awx/main/tests/settings_for_test.py:16`), so `cleanup_jobs()` cannot be called end to end in a test at all. That is why this went unnoticed, and it is worth its own look.

## Upstream

`devel` carries the identical line, so the same fix is proposed there.

### CODE REVIEW

File: awx/main/tests/functional/commands/test_cleanup_jobs.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    """A batch window covering only ids whose jobs are already gone must not fail.
+
+    Job shares the UnifiedJob id sequence with project updates, inventory
+    updates and workflow jobs, so a window of ids holding no deletable Job is
+    routine rather than exceptional.
+    """
+    Job.objects.filter(pk__in=[old_jobs[2].pk, old_jobs[3].pk]).delete()
+    expected = Job.objects.filter(pk__in=[job.pk for job in old_jobs]).count()
+
+    skipped, deleted = _command(batch_size=1).cleanup_jobs()
```
[the remark this answers is not in the document]
Comment (blaipr) [reply awx/main/tests/functional/commands/test_cleanup_jobs.py:57]: This will not fail lint. Pyflakes only reports F841 for a plain assignment, not for a tuple unpacking target, and `api-lint` was green on the commit this comment was left against.

You are right that the value was going unused though. I have asserted on it instead of renaming it, since `skipped == 0` is the part of the return contract these tests actually care about.

**Comment by blaipr** [comment]
Pushed 8fccb9d, which asserts `skipped == 0` in the three tests that were unpacking the value and dropping it. The skipped count is every job newer than the cutoff or still pending, waiting or running, none of which the fixture creates, so zero is what these tests mean when they say the window held nothing deletable.

The review also flagged the unused name as an F841 lint failure. It is not one, pyflakes only reports F841 for a plain assignment and not for a tuple unpacking target, and `api-lint` was green on the commit it was raised against. Asserting on the value seemed better than renaming it to `_` either way.

**Review by cigamit — APPROVED** [review]
